Client/Frontend/NTP/NTPLearnMoreContentView.swift

```diff
@@ -0,0 +1,226 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+import BraveShared
+import BraveRewardsUI
+
+extension NTPLearnMoreViewController {
+    enum NTPButtonType {
+        case rewards, ads
+    }
+
+    struct NTPNotificationLearnMoreViewConfig {
+        let headerText: String
+        let buttonType: NTPButtonType?
+        let tosText: Bool
+        let learnMoreButtonText: String
+    }
+    
+    class NTPLearnMoreContentView: UIStackView {
```

Comment:
Most of this view was previously abstracted so both the dark background and the white modal view had the same view logic.
Now this is separate class since the design team wanted changes for the modal view only.